jmxtrans-cluster/src/test/java/com/googlecode/jmxtrans/cluster/ClusterServiceTest.java

```diff
@@ -0,0 +1,118 @@
+
+/**
+ * ClusterService Tester.
+ *
+ * @author Tibor Kulcsar
+ * @since <pre>May 14, 2016</pre>
+ */
+@Category({IntegrationTest.class, RequiresIO.class})
+@AllowDNSResolution
+@AllowNetworkListen(ports = {0})
+@AllowNetworkAccess(endpoints = {"127.0.0.1:*"})
+@AllowLocalFileAccess(paths = {"*"})
+public class ClusterServiceTest {
+
+    private static TestingServer testingServer;
+    private static CuratorFramework client;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        testingServer = new TestingServer(2181,true);
+        //client = CuratorFrameworkFactory.newClient("10.0.0.6:2181", new ExponentialBackoffRetry(1000,3));
+        client = CuratorFrameworkFactory.newClient(testingServer.getConnectString(), new ExponentialBackoffRetry(1000,3));
+        //client = CuratorFrameworkFactory.newClient("10.189.33.100:2181,10.189.33.101:2181,10.189.33.102:2181", new ExponentialBackoffRetry(1000,3));
+        client.start();
+    }
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        testingServer.stop();
+    }
+
+    @Test
+    public void testZookeeperConnectionStartup() throws Exception {
+        ClusterService service = ClusterServiceFactory.createClusterService(
+                TestUtils.createGoldenConfiguration(testingServer.getConnectString()));
+        service.startService();
+
+        synchronized((ZookeeperClusterService)service){
+            try{
+                (service).wait();
+            }catch(InterruptedException e){
+                e.printStackTrace();
+            }
+        }
+
+        Stat stat = client.checkExists().forPath("/jmxtrans/workers/worker_01");
+        assertThat(stat).isNotEqualTo(null);
+
+        service.stopService();
+    }
+
+    @Test
+    public void testConfigUpdate() throws Exception{
+
+        client.create().creatingParentContainersIfNeeded()
+                .withMode(CreateMode.PERSISTENT)
+                .forPath("/jmxtrans/jvms/jvm_01/config", "Config01".getBytes());
+        client.create().creatingParentContainersIfNeeded()
+                .withMode(CreateMode.PERSISTENT)
+                .forPath("/jmxtrans/jvms/jvm_01/affinity", "worker_01".getBytes());
+        client.create().creatingParentContainersIfNeeded()
+                .withMode(CreateMode.PERSISTENT)
+                .forPath("/jmxtrans/jvms/jvm_01/owner");
+        client.create().creatingParentContainersIfNeeded()
+                .withMode(CreateMode.PERSISTENT)
+                .forPath("/jmxtrans/jvms/jvm_02/config", "Config02".getBytes());
+        client.create().creatingParentContainersIfNeeded()
+                .withMode(CreateMode.PERSISTENT)
+                .forPath("/jmxtrans/jvms/jvm_02/affinity", "worker_02".getBytes());
+        client.create().creatingParentContainersIfNeeded()
+                .withMode(CreateMode.PERSISTENT)
+                .forPath("/jmxtrans/jvms/jvm_02/owner");
+        Thread.sleep(3000);
```

Comment:
`Thread.sleep()` in tests is problematic. Any chance we could do proper synchronization? Or at least use something like [awaitility](https://github.com/jayway/awaitility)?